js/src/ui/TxList/store.js

```diff
@@ -14,150 +14,121 @@
-          ]) => {
-            return {
-              ...transactionReceipt,
-              ...transaction
-            };
-          });
-      });
+  loadTransactions (_txhashes) {
+    const { eth } = this._api;
 
-    if (!promises.length) {
+    // Ignore special cases and if the contents of _txhashes && this.sortedHashes are the same
+    if (Array.isArray(_txhashes) || this.sameHashList(_txhashes)) {
+      return;
+    }
+    // If there was a hash removed:
+    if (_txhashes.length < this.sortedHashes.length) {
+      this.removeHash(_txhashes);
       return;
     }
 
-    Promise
-      .all(promises)
-      .then((_transactions) => {
-        const blockNumbers = [];
-        const transactions = _transactions
-          .filter((tx) => tx && tx.hash)
-          .reduce((txs, tx) => {
-            txs[tx.hash] = tx;
-
-            if (tx.blockNumber && tx.blockNumber.gt(0)) {
-              blockNumbers.push(tx.blockNumber.toNumber());
+    _txhashes
+      .map((txhash) => {
+        eth.getTransactionByHash(txhash)
+          .then((tx) => {
+            this.transactions[txhash] = tx;
+            // If the tx has a blockHash, let's get the blockNumber, otherwise it's ready to be added
+            if (tx.blockHash) {
+              eth.getBlockByNumber(tx.blockNumber)
+                .then((block) => {
+                  this.blocks[tx.blockNumber] = block;
+                  this.addHash(txhash);
+                });
+            } else {
+              this.addHash(txhash);
             }
-
-            return txs;
-          }, {});
-
-        // No need to add transactions if there are none
-        if (Object.keys(transactions).length === 0) {
-          return false;
-        }
-
-        this.addTransactions(transactions);
-        this.loadBlocks(blockNumbers);
-      })
-      .catch((error) => {
-        console.warn('loadTransactions', error);
+          });
       });
   }
 
-  loadBlocks (_blockNumbers) {
-    const blockNumbers = uniq(_blockNumbers).filter((bn) => !this.blocks[bn]);
+  cancelTransaction = (txComponent, tx) => {
+    const { parity } = this._api;
+    const { hash } = tx;
 
-    if (!blockNumbers || !blockNumbers.length) {
-      return;
-    }
+    parity.removeTransaction(hash)
+    .then(() => {
+      txComponent.setState({ canceled: true });
+    })
+    .catch((err) => {
+      this._newError({ message: err });
+    });
+  }
 
-    Promise
-      .all(blockNumbers.map((blockNumber) => this._api.eth.getBlockByNumber(blockNumber)))
-      .then((blocks) => {
-        this.addBlocks(
-          blocks.reduce((blocks, block, index) => {
-            blocks[blockNumbers[index]] = block;
-            return blocks;
-          }, {})
-        );
+  editTransaction = (txComponent, tx) => {
+    const { parity } = this._api;
+    const { hash, gas, gasPrice, to, from, value, input, condition } = tx;
+
+    parity.removeTransaction(hash)
+      .then(() => {
+        parity.postTransaction({
```

Comment:
Indentation of this block and the containing one. In this case (and I generally do exactly what you've done myself), it could aid readability to have `transaction` as a `const` and passing it in.